### PR TITLE
fix: write lifecycle hook enable flag & custom postinstall to aspect_rules_js.json instead of overwriting package.json

### PR DESCRIPTION
**Pull request by gregmagolan**
The current behavior of writing to package.json ends up re-ordering its fields which can break modules resolution if `exports` field is re-ordered.

See https://github.com/aspect-build/rules_js/issues/385#issuecomment-1217510270 for more context.

### CODE REVIEW

File: npm/private/lifecycle/lifecycle-hooks.js [hunk 1]
```diff
@@ -110,22 +110,20 @@ async function main(args) {
         unsafePerm: true, // Don't run under a specific user/group
     }
 
-    const packageJson = JSON.parse(
-        await fs.promises.readFile(path.join(packageDir, 'package.json'))
+    const rulesJsJson = JSON.parse(
```
Comment (kormide) [inline npm/private/lifecycle/lifecycle-hooks.js:113]: `rulesJsOptions`?

Comment (kormide) [reply npm/private/lifecycle/lifecycle-hooks.js:113]: `lifecycleOptions`?

File: npm/private/lifecycle/lifecycle-hooks.js [hunk 2]
```diff
@@ -110,22 +110,20 @@ async function main(args) {
...
+        await fs.promises.readFile(
+            path.join(packageDir, 'aspect_rules_js.json')
```
Comment (kormide) [inline npm/private/lifecycle/lifecycle-hooks.js:115]: `aspect_rules_js_lifecycle.json`?

**Review by kormide — APPROVED** [review]
LGTM. Just left a few thoughts on naming.

File: npm/private/lifecycle/lifecycle-hooks.js (hunk 2 above)
Comment (gregmagolan) [reply npm/private/lifecycle/lifecycle-hooks.js:115]: we might want to put other metadata in here besides lifecycles

File: npm/private/lifecycle/lifecycle-hooks.js (hunk 1 above)
Comment (gregmagolan) [reply npm/private/lifecycle/lifecycle-hooks.js:113]: 👍

**Pull request merged**